src/main/cc/wfanet/panelmatch/common/crypto/cryptor.h

```diff
@@ -0,0 +1,57 @@
+// Copyright 2020 The Cross-Media Measurement Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef SRC_MAIN_CC_WFANET_PANELMATCH_COMMON_CRYPTO_CRYPTOR_H_
+#define SRC_MAIN_CC_WFANET_PANELMATCH_COMMON_CRYPTO_CRYPTOR_H_
+
+#include <memory>
+#include <string>
+
+#include "absl/status/status.h"
+#include "absl/status/statusor.h"
+#include "absl/strings/string_view.h"
+#include "absl/types/span.h"
+
+#include "crypto/ec_commutative_cipher.h"
+
+namespace wfanet::panelmatch::common::crypto {
+using ::private_join_and_compute::ECCommutativeCipher;
+// A cryptor dealing with basic operations needed for panel match
+class Cryptor {
+ public:
+  virtual ~Cryptor() = default;
+
+  Cryptor(Cryptor&& other) = delete;
+  Cryptor& operator=(Cryptor&& other) = delete;
+  Cryptor(const Cryptor&) = delete;
+  Cryptor& operator=(const Cryptor&) = delete;
+
+  virtual absl::StatusOr<std::string> Decrypt(
```

Comment:
What do you think about supporting a batch interface here to avoid needing to re-acquire the Mutex between each low-level operation? Maybe `absl::StatusOr<std::vector<std::string>> Whatevercrypt(absl::Span<const absl::string_view> plaintexts_or_ciphertexts)`?

Reply:
That sounds great and makes sense. I'll add that in.

Reply:
@efoxepstein is there a standard length of a plaintext?

Reply:
In some sense it doesn't matter -- what happens under the hood is:

1. Any string is hashed to point on an elliptic curve
2. The point itself is encrypted
3. The hash itself isn't reversible (but since hashing is deterministic, we can [but don't actually need to] build lookup tables)

We expect the plaintexts to be email addresses, user ids, etc.